### PR TITLE
Add a component for prefetching images

### PR DESCRIPTION
**Pull request by creativecreature**
## Overview
Gameface haven't implemented the `onLoad` event for images. Therefore, it's difficult to make a smooth UI where images don't pop in randomly as they are being fetched.

In this PR I've added a component that can prefetch images and invoke a callback when that is done. At that point they should be cached by the browser, and it should be a lot easier to have a smooth effect in other parts of the application.

### CODE REVIEW

**Comment by marlonicus** [comment]
Smart solution to the lack of `onLoad`! 

I am wondering if we should not expose this as a new component, but instead integrate the functionality into the `Image` component so we can leverage its existing functionality.

**Review by pirelenito — CHANGES_REQUESTED** [review]
Awesome stuff! My only suggestion is to move this new component away from `@react-facet/core` as that package has nothing that is "Gameface specific".

My recommendation is to create a new package to hold this component, such as `@react-facet/gameface-image-preloader`.

Thoughts?

**Comment by pirelenito** [comment]
@hebbeh reached out to me regarding a concern on adding this new package under the `@react-facet` namespace. After some chat we've agreed that it is probably a good idea to have this under a `@ore-ui` namespace.

So the final suggestion to the PR is to rename the package to `@ore-ui/gameface-image-preloader`.

Any thoughts @xaviervia @creativecreature ?

File: packages/@ore-ui/gameface-image-preloader/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# @ore-ui/gameface-image-preloader
+There isn't support for the onLoad event in Gameface right now. Therefore, it's hard to create a smooth UI where
+external images don't "pop in". This component solves that by rendering an invisible list of images, and invoking a
+callback when all of them have received a height and width. This should, hopefully, mean that they have been cached by
+the browser - which in turn should enable subsequent renders within the same animation frame.
+
+This package is part of ore-ui(TODO: LINK), and uses @react-facet(TODO: LINK) for it's state management.
```
Comment (pirelenito) [inline packages/@ore-ui/gameface-image-preloader/README.md:7]: Add missing links.

File: packages/@ore-ui/gameface-image-preloader/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# @ore-ui/gameface-image-preloader
+There isn't support for the onLoad event in Gameface right now. Therefore, it's hard to create a smooth UI where
```
Comment (pirelenito) [inline packages/@ore-ui/gameface-image-preloader/README.md:2]: Formatting: we probably want a line break here. (unfortunately we don't have a prettier check yet for Readme files).

Comment (creativecreature) [reply packages/@ore-ui/gameface-image-preloader/README.md:2]: I had a look at the other README files, and it didn't look like we followed any convention for line length. The root README has a line with more than 300 characters. Therefore, I originally set the text width to 120 (same as the code). I have now set it to 80 and reformatted this file.

**Comment by creativecreature** [comment]
@hebbeh  has sent a feature request to Coherent (to ask for the onLoad event). Therefore, we'll pause this PR until we have received a response from them. If it is on their roadmap, and will be implemented within a reasonable timeframe, we'll close this PR and use that event instead.

**Pull request closed without merging**